internal/handlers/common/compare.go

```diff
@@ -193,10 +193,31 @@ func compare(docValue, filter any) compareResult {
 
 	switch docValue := docValue.(type) {
 	case *types.Document:
-		// TODO: implement document comparing
+		if filter, ok := filter.(*types.Document); ok {
+			switch compareDocuments(filter, docValue) {
```

Comment:
here you can probably return `compareResults`, aren't you? 
and linter check will pass.